src/func.d

```diff
@@ -3847,29 +3813,43 @@ extern (C++) int overloadApply(Dsymbol fstart, void* param, int function(void*,
             if (next == fstart)
                 break;
         }
-        else if (TemplateDeclaration td = d.isTemplateDeclaration())
+        else if (auto td = d.isTemplateDeclaration())
         {
             if (int r = (*fp)(param, td))
                 return r;
             next = td.overnext;
         }
-        else
+        else if (auto fd = d.isFuncDeclaration())
         {
-            FuncDeclaration fd = d.isFuncDeclaration();
-            if (!fd)
-            {
-                d.error("is aliased to a function");
-                break;
-                // BUG: should print error message?
-            }
             if (int r = (*fp)(param, fd))
                 return r;
             next = fd.overnext;
         }
+        else
+        {
+            d.error("is aliased to a function");
+            break;
+            // BUG: should print error message?
+        }
     }
     return 0;
 }
 
+extern (D) int overloadApply(Dsymbol fstart, scope int delegate(Dsymbol) dg)
```

Comment:
IIUC the D version is wrapping the C++ version.  Instead, could the extern(C++) version wrap the D version?